### DISCUSSION TITLE
lit@2.2.0 interfering with custom pointermove mixin

### DISCUSSION THREAD

**Discussion by patorpey** [lit/lit#2581] [Q&A] [answered]
I experienced an issue when updating from lit@2.1.4 to 2.2.0 that broke a good deal of functionality within my application. However, I cannot isolate what exactly is causing the issue, so cannot file a proper bug issue. I'm hoping that this discussion thread might provide some additional insights into the changes between 2.1.4 and 2.2.0 that can help me isolate the issue to report or create a minimal reproduction.

The application in question uses a mixin of sorts to handle drag-move operations on "draggable" elements (some have shadow roots, some do not). It listens to `pointerdown` events on a container element, manages adding and removing `pointermove` handlers, and generates a custom `pointerdrag` event (bubbles and composed). This code predated my use of lit and has worked fine with LitElement since before lit 2.0. (It's on my backlog to see if I can turn it into a reactive controller.)

After upgrading to lit@2.2.0, I see the `pointerdown` events on my elements, but `pointermove` events are somehow getting lost. If I roll back to lit@2.1.4, things work as expected. I've observed this behaviour in both Chrome 98 and Firefox 97.0.1.

I see that 2.2.0 adds some custom debug events. I tried setting `window.emitLitDebugLogEvents` to `true` and `false`, which had no affect on this issue. I've looked at the diffs for the lit release and nothing stood out to me as being potentially related to my issue, but, not being familiar with the lit-element and lit-html source, it's difficult to tell.

I'm hoping someone might have encountered something similar or someone more familiar with the 2.2.0 changes might have advice on how to isolate the issue so that I can implement a workaround or report an issue.

**Comment by rictic** [comment, marked as answer]
This is definitely unexpected. The new devlog event system should be a 100% noop if `window.emitLitDebugLogEvents` isn't true, and there isn't anything else that came in with 2.2.0.

Is there a possibility that anything else changed between 2.1.4 and 2.2.0?

**Reply by patorpey** [reply]
I thought that the issue might be setting `window.emitLitDebugEvents` from the console, perhaps after some initialization. I tried setting `window.emitLitDebugEvents = true` as early as possible in my code, but still have the issue.

I tried to isolate the issue to lit by running `rm -rf node_modules` and `npm install` with lit@2.1.4 and my application runs correctly. If `npm i lit@2.2.0` and rebuild and run, I experience the problem with the pointermove events described above. If I reset my `package.json` to have `"lit": "~2.1.4"`, clear `node_modules`, and `npm i` again, the correct behaviour returns. So, it's definitely some interaction with the changes in 2.2.0. I'll stay with ~2.1.4 for now, but hope I'm not stuck there as Lit evolves.

**Reply by rictic** [reply]
Please give v2.2.1 a try, we removed a problematic case where we were cloning nodes as part of creating debug events, which can cause unexpected behavior as some constructors were being called for cloned custom elements.

**Reply by patorpey** [reply]
@rictic 2.2.1 appears to have resolved the issue for me. Thank you!
